core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2350,13 +2354,12 @@ class SparkContext(config: SparkConf) extends Logging {
     try {
       val listenerClassNames: Seq[String] =
         conf.get("spark.extraListeners", "").split(',').map(_.trim).filter(_ != "")
-      for (className <- listenerClassNames) {
-        // Use reflection to find the right constructor
+      val extraListeners = listenerClassNames.map{ className =>
```

Comment:
`.map { className =>`

You have a lot of style issues in your code - indentation, spacing, etc. Please read the style section in http://spark.apache.org/contributing.html and try to follow it.